key_gen.go

```diff
@@ -57,17 +79,14 @@ func KeyForPrefixedStringMap(
 	return keyForPrefixedStringMaps(prefix, stringMap)
 }
 
-// keyForPrefixedStringMaps generates a unique key for a prefix and a series
-// of maps containing tags.
-//
-// If a key occurs in multiple maps, keys on the right take precedence.
-func keyForPrefixedStringMaps(prefix string, maps ...map[string]string) string {
-	keys := keyGenPool.stringsPool.Get().([]string)
+func keyForPrefixedStringMapsAsKey(prefix string, maps ...map[string]string) key {
+	keys := keyGenPool.stringsPool.Get()
 	for _, m := range maps {
 		for k := range m {
 			keys = append(keys, k)
 		}
 	}
+	// 1 allocation as sort.Interface escapes to heap
 	sort.Strings(keys)
 
 	buf := keyGenPool.bufferPool.Get().(*bytes.Buffer)
```

Comment:
```suggestion
	buf := make([]byte, 0, 256)
```
If you similarly use a stack-allocated byte-slice, you avoid the same problems as with the string slice (note the size 256 it just a suggestion). 

For the `WriteString` calls below you can use `buf = append(buf, prefix...)` to write out the content of the string to the byte slice in a non-allocating fashion.

Reply:
Done. Using a stack allocated byte slice.